api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -50,52 +51,56 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	 */
 	@Override
 	public Cohort getCohort(Integer id) throws DAOException {
-		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
+		return sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
 
 	/**
 	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(Integer, boolean, Date)
 	 */
 	@Override
-	@SuppressWarnings("unchecked")
 	public List<Cohort> getCohortsContainingPatientId(Integer patientId, boolean includeVoided,
-	                                                  Date asOfDate) throws DAOException {
-		Disjunction orEndDate = Restrictions.disjunction();
-		orEndDate.add(Restrictions.isNull("m.endDate"));
-		orEndDate.add(Restrictions.gt("m.endDate", asOfDate));
+													  Date asOfDate) throws DAOException {
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<Cohort> query = cb.createQuery(Cohort.class);
+		Root<Cohort> root = query.from(Cohort.class);
+
+		Join<Cohort, CohortMembership> membershipJoin = root.join("memberships");
+
+		List<Predicate> predicates = new ArrayList<>();
 
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Cohort.class);
-		criteria.createAlias("memberships", "m");
 		if (asOfDate != null) {
-			criteria.add(Restrictions.le("m.startDate", asOfDate));
-			criteria.add(orEndDate);
+			predicates.add(cb.lessThanOrEqualTo(membershipJoin.get("startDate"), asOfDate));
+
+			Predicate endDateNullPredicate = cb.isNull(membershipJoin.get("endDate"));
+			Predicate endDateGtPredicate = cb.greaterThan(membershipJoin.get("endDate"), asOfDate);
+			predicates.add(cb.or(endDateNullPredicate, endDateGtPredicate));
 		}
-		criteria.add(Restrictions.eq("m.patientId", patientId));
-		criteria.setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY);
-		
+		predicates.add(cb.equal(membershipJoin.get("patientId"), patientId));
+
 		if (!includeVoided) {
-			criteria.add(Restrictions.eq(VOIDED, includeVoided));
+			predicates.add(cb.equal(root.get(VOIDED), includeVoided));
 		}
-		return criteria.list();
-	}
 
+		query.distinct(true).where(predicates.toArray(new Predicate[0]));
```

Comment:
How about using this? `predicates.toArray(new Predicate[]{})`

Reply:
I wasn't familiar with that syntax but I agree it is more readable, updated.